llpc/util/llpcCacheAccessor.cpp

```diff
@@ -60,24 +65,108 @@ CacheAccessor::CacheAccessor(Context *context, MetroHash::Hash &cacheHash, Compi
 // @param hash : The hash for the entry to access.
 // @param userCache : The ICache supplied by the application. nullptr if no cache is provided.
 // @param userShaderCache : The shader cache supplied by the application. nullptr if no cache is provided.
-// @param compiler : The compiler object with the internal caches.
+// @param internalCaches : The internal caches to check.
 void CacheAccessor::initialize(MetroHash::Hash &hash, Vkgc::ICache *userCache, IShaderCache *userShaderCache,
-                               Compiler *compiler) {
-  assert(compiler);
-  m_compiler = compiler;
-  m_userCache = userCache;
-  m_userShaderCache = userShaderCache;
-  m_shaderCache = nullptr;
-  m_shaderCacheEntryState = ShaderEntryState::New;
+                               CachePair internalCaches) {
+  m_internalCaches = internalCaches;
+  m_applicationCaches = {userCache, userShaderCache};
+  ResetShaderCacheTrackingData();
   m_elf = {0, nullptr};
 
+  lookUpInICaches(hash);
+  if (m_cacheResult != Result::Success)
+    lookUpInShaderCaches(hash);
+}
+
+// =====================================================================================================================
+// Looks for the given hash in the ICaches and sets the cache accessor state with the results.
+//
+// @param hash : The hash to look up.
+void CacheAccessor::lookUpInICaches(const MetroHash::Hash &hash) {
```

Comment:
lookUpInICaches -> lookUpInCaches
I think using ICaches in the function name is somewhat strange.